Sources/Sham/Models/StubRequest.swift

```diff
@@ -110,17 +110,46 @@ public struct StubRequest: CustomStringConvertible {
         
         return validScheme && validHost && validPort && validPath && validQuery
     }
-}
-
-// MARK: - Hashable and Equatable
-
-extension StubRequest: Hashable, Equatable {
-    public func hash(into hasher: inout Hasher) {
-        hasher.combine(self.description)
-    }
     
-    public static func == (lhs: Self, rhs: Self) -> Bool {
-        return lhs.hashValue == rhs.hashValue
+    /// A function that generates a score to represent how well a stored stubbed request matches an incoming request
+    /// - Parameter request: The incoming request to generate the priority score against
+    /// - Returns: A score representing how well this request matches an incoming request.  Nil values imply no match.
+    public func priorityScore(for request: StubRequest) -> Int? {
+        guard self.canMockData(for: request) else {
+            return nil
+        }
+        
+        var score = 0
```

Comment:
With a weighting function, we don't want to return nil, we could instead return `0` to represent it has no match. A `0` making it through to the end of the function implies that it doesn't even have a single matching part of the URL at all, which is a good enough indicator. We probably don't even need to check `canMockData` here. Thoughts?

If this changes, the comments above need correction also.

```suggestion
    public func priorityScore(for request: StubRequest) -> Int {
        var score = 0
    
        guard self.canMockData(for: request) else {
            return 0
        }
```

Reply:
I added the `canMockData` call because it includes additional logic the priority score function wasnt handling, seemed like a decent check.